homeassistant/components/launch_library/__init__.py

```diff
@@ -1 +1,56 @@
 """The launch_library component."""
+from datetime import timedelta
+import logging
+
+from pylaunches import PyLaunches, PyLaunchesException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, PLATFORMS, UPDATECOORDINATOR
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
+
+    if hass.data.get(DOMAIN) is None:
+        hass.data.setdefault(DOMAIN, {})
+
+    session = async_get_clientsession(hass)
+    launches = PyLaunches(session)
+
+    async def async_update():
+        try:
+            return await launches.upcoming_launches()
+        except PyLaunchesException as ex:
+            raise UpdateFailed(ex) from ex
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=DOMAIN,
+        update_method=async_update,
+        update_interval=timedelta(minutes=30),
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data[DOMAIN][entry.entry_id] = {
+        UPDATECOORDINATOR: coordinator,
+    }
```

Comment:
```suggestion
    hass.data[DOMAIN] = coordinator
```

Only the coordinator is stored, and we can't set this up multiple times